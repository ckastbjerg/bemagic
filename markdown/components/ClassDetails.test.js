/* eslint-env jest */
const ClassDetails = require('./ClassDetails');

test('it throws', () => {
    expect(() => ClassDetails()).toThrow();
});

test('it works', () => {
    expect(ClassDetails({
        atRules: {
            description: 'Some description...',
            intro: 'Some intro',
            tag: 'span',
        },
        name: 'block',
        markup: '<div class="block">...</div>',
        states: ['hover', 'active', 'disabled'],
        themes: ['inverse', 'secondary'],
    })).toMatchSnapshot();
});
