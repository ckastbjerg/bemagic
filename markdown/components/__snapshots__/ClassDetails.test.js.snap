exports[`test it works 1`] = `
"
        <a name=block></a>
        #### \`block\` 
        > Some description...
        <div class=\"block\">...</div>
        Elements using this class will change appearance when used inside [theme]
        (../CONVENTIONS.md#theming) \`inverse\` or \`secondary\`.
        The component will react to \`hover\`, \`active\` and \`disabled\`. states
        ###### Specifications

        - **Tag:** span
        ----------
    "
`;
